sklearn/cluster/_kmeans.py

```diff
@@ -496,15 +513,25 @@ def _kmeans_single_lloyd(X, sample_weight, centers_init, max_iter=300,
 
             centers, centers_new = centers_new, centers
 
-            center_shift_tot = (center_shift**2).sum()
-            if center_shift_tot <= tol:
+            if np.array_equal(labels, labels_old):
+                # First check the labels for strict convergence because
+                # center_shift might not be exactly 0 due to rounding errors.
```

Comment:
This comment needs to be updated since this branch is now called even if `tol > 0`.